library/std/src/sys/net/connection/uefi/tcp4.rs

```diff
@@ -88,6 +88,44 @@ impl Tcp4 {
         }
     }
 
+    pub(crate) fn write(&self, buf: &[u8]) -> io::Result<usize> {
+        let evt = unsafe { self.create_evt() }?;
+        let completion_token =
+            tcp4::CompletionToken { event: evt.as_ptr(), status: Status::SUCCESS };
+        let data_len = crate::cmp::min(u32::MAX as u64, buf.len() as u64) as u32;
+
+        let fragment = tcp4::FragmentData {
+            fragment_length: data_len,
+            fragment_buffer: buf.as_ptr() as *mut _,
```

Comment:
Nit to make the cast obvious (especially since it includes a mutability change)

```rust
            fragment_buffer: buf.as_ptr().cast::<c_void>().cast_mut()
```